### PR TITLE
Load YouTube assets on fronts on placeholder click

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Currently, YouTube assets for all videos on a front are loaded up front with the enhanced JavaScript, regardless of whether the user intends to watch any videos. 

This change loads the YouTube assets only when the user clicks on the video placeholder. When the video loads, it immediately starts playing ([`player.playVideo()`](https://developers.google.com/youtube/iframe_api_reference#playVideo) is called programatically)

This does not affect the behaviour of video articles or videos embedded in articles.

## Screenshots

**Before**

screen shot 2018-09-05 at 14 27 17

**After**

screen shot 2018-09-05 at 14 26 43

## What is the value of this and can you measure success?

I will flesh this out with some actual numbers, but in general:

- Far fewer requests to YouTube (~30 on the video front)
- Fewer unnecessary assets loaded (~3MB on the video front)
- Shorter time to interactive (TBC)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Accessibility test checklist

- [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/) - still bad
- [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/) - still bad
- [x] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by paperboyo** [comment]
💎!

File: common/app/views/fragments/atoms/youtube.scala.html [hunk 1]
```diff
@@ -77,7 +77,7 @@
                 </div>
             }
             @bestPosterImage.map { image =>
-                <div class="@RenderClasses(Map("youtube-media-atom__overlay" -> true, "vjs-big-play-button" -> !expired))" style="background-image: url(@Video700.bestSrcFor(image))">
+                <div class="@RenderClasses(Map("youtube-media-atom__overlay" -> true, "vjs-big-play-button" -> !expired))" style="background-image: url(@Video700.bestSrcFor(image))" role="button">
```
Comment (SiAdcock) [inline common/app/views/fragments/atoms/youtube.scala.html]: I think this is adding noise and not achieving any real value, I’ll probably back it out

File: static/src/stylesheets/module/atoms/_youtube.scss [hunk 2]
```diff
@@ -102,7 +102,7 @@ youtube-media-atom:not(.youtube-related-videos) .youtube-media-atom__iframe.yout
     }
 }
 
-.youtube-media-atom__overlay.vjs-big-play-button {
+.content .youtube-media-atom__overlay.vjs-big-play-button {
```
Comment (SiAdcock) [inline static/src/stylesheets/module/atoms/_youtube.scss]: Maybe needs a comment?

**Comment by SiAdcock** [comment]
For consideration:

- autoplay might be blocked by the browser in certain situations, requiring a further click / tap?
- autoplay does not count as a play in YouTube insights 
- are plays / progress still being tracked in Ophan?

**Comment by paperboyo** [comment]
> Can we consider this an acceptable trade-off against the performance improvement for the vast majority of users who don’t watch videos in fronts?

I vote yes FWIW. Even more so that the delay seems lower if some of it is already in user’s cache.

File: static/src/javascripts/projects/common/modules/atoms/youtube.js [hunk 3]
```diff
@@ -177,11 +177,14 @@ const shouldAutoplay = (atomId: string): boolean => {
     const flashingElementsAllowed = () =>
         accessibilityIsOn('flashing-elements');
 
+    const isVideoArticle = () => config.get('page.contentType') === 'Video';
```
Comment (akash1810) [inline static/src/javascripts/projects/common/modules/atoms/youtube.js]: Would it be safer to compare the `toLowercase` of `config.get`?

**Review by akash1810 — APPROVED** [review]

**Comment by akash1810** [comment]
> there is an increased delay between clicking the placeholder and seeing the video as the YouTube assets are loaded.

hopefully this is only on the first video as for the second one the browser has cached youtube's assets? 🤞

**Comment by SiAdcock** [comment]
To be on the safe side, I'm going to merge this on Monday, and email user help to look out for reports of problems playing videos on fronts.

**Review by AWare — APPROVED** [review]

**Comment by mchv** [comment]
💎

**Comment by SiAdcock** [comment]
@addyosmani's lovely [overview of performance budgets](https://addyosmani.com/blog/performance-budgets/) uses our network front as an example. I believe the huge drop in JS size in this graph is entirely due to this change 👌

speedcurve-guardian

**Pull request merged**